zhaquirks/sonoff/button.py

```diff
@@ -1,70 +1,47 @@
-"""Device handler for eWeLink WB01."""
-from zigpy.profiles import zha
-from zigpy.quirks import CustomDevice
-from zigpy.zcl.clusters.general import Basic, Identify, OnOff, PowerConfiguration
+"""Device handler for Sonoff buttons."""
+from zigpy.quirks.v2 import add_to_registry_v2
 
 from zhaquirks.const import (
     BUTTON,
+    CLUSTER_ID,
     COMMAND,
     COMMAND_OFF,
     COMMAND_ON,
     COMMAND_TOGGLE,
-    DEVICE_TYPE,
     DOUBLE_PRESS,
-    ENDPOINTS,
-    INPUT_CLUSTERS,
+    ENDPOINT_ID,
     LONG_PRESS,
-    MODELS_INFO,
-    OUTPUT_CLUSTERS,
-    PROFILE_ID,
     SHORT_PRESS,
 )
 
+(
+    add_to_registry_v2("eWeLink", "WB01").device_automation_triggers(
+        {
+            (SHORT_PRESS, BUTTON): {COMMAND: COMMAND_TOGGLE},
+            (DOUBLE_PRESS, BUTTON): {COMMAND: COMMAND_ON},
+            (LONG_PRESS, BUTTON): {COMMAND: COMMAND_OFF},
+        }
+    )
+)
 
-class SonoffButton(CustomDevice):
-    """Custom device representing sonoff devices."""
-
-    signature = {
-        # <SimpleDescriptor endpoint=1 profile=260 device_type=0 device_version=0
-        # input_clusters=[0, 3, 1] output_clusters=[6, 3]>
-        MODELS_INFO: [("eWeLink", "WB01")],
-        ENDPOINTS: {
-            1: {
-                PROFILE_ID: zha.PROFILE_ID,
-                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    PowerConfiguration.cluster_id,
-                    Identify.cluster_id,
-                ],
-                OUTPUT_CLUSTERS: [
-                    Identify.cluster_id,
-                    OnOff.cluster_id,
-                ],
+(
+    add_to_registry_v2("eWeLink", "SNZB-01P").device_automation_triggers(
+        {
+            (SHORT_PRESS, BUTTON): {
+                COMMAND: COMMAND_TOGGLE,
+                CLUSTER_ID: 6,
+                ENDPOINT_ID: 1,
```

Comment:
Diff looks bad for all changes, but we can probably combine the two different Sonoff button v2 quirks, as they essentially use the same device triggers. Currently, one quirk just narrows it down to endpoint and cluster which we probably don't need.

Reply:
https://github.com/zigpy/zha-device-handlers/pull/3050/commits/55aa0f270c3796d38f5992113502e7dbf8b13a80